library/ajax/sql_server_status.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * sql_server_status.php
+ *
+ * @package   OpenEMR
+ * @link      https://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2020 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+/*
+ * I wrote this mainly to show server activity for transaction intensive upgrades
+ * where the user can know we are still working though no activity from upgrade sequence.
+ * Not sure I can do csrf here!
+ * */
+
+$ignoreAuth = true;
```

Comment:
Regarding security, couldn't a bad actor just keep polling this script over web and collect data?

Gotta be a way to secure this, but too late to think clearly over here :)

Reply:
yep, I'll get a csrf working for it. Polling has issues still where I may need to rethink some of it. Turning it off is proving a little stubborn.

Reply:
If the csrf token is provided to the sql_upgrade.php script, though, then would just take a 1-2 punch to run this script (run sql_upgrade.php and scrape the csrf) to heart's content without auth.
We gotta figure out a way, though, to get this concept in. May need to figure out how to limit scope or where the calls originate from or go to, though.

Reply:
Okay, i'm dwelling on it. Now that you've reminded me about legacy, makes it a little more difficult!
I'm now thinking that we add a flag to upgrade script to only use this feature for what we know to be long running upgrades.
And only then give user a button to take a snapshot of server activity if concerned where upgrade is activity wise.

Reply:
Note in this case, we could deal with legacy by adding anything needed to the database foorprint upfront. For example, if decided to store/retrieve from a new longtext field in version table called upgrade_status, then would just add this upfront if does not yet exist.